File: array_count.js
const array = [
  "apple",
  "potato",
  "apple",
  "orange",
  "lemon",
  "apple",
  "nut",
  "lemon",
];

const count = array.length;
const map = {};
for (let i = 0; i < count; i++) {
  map[array[i]] = (map[array[i]] || 0) + 1;
}

console.log(map);

File: array_count.go
package main

import (
	"fmt"
)

var array = []string{
	"apple",
	"potato",
	"apple",
	"orange",
	"lemon",
	"apple",
	"nut",
	"lemon",
}

func main() {
	frequency := make(map[string]int)
	for i := 0; i < len(array); i++ {
		frequency[array[i]]++
	}
	// fmt.Println(array)
	fmt.Println(frequency)
}
